single-point-processing-example/index.js

```diff
@@ -1,13 +1,44 @@
+const { RekognitionClient, DetectLabelsCommand } = require("@aws-sdk/client-rekognition")
 
 /**
  * Used to identify if a PNG is a dog or cat
  *
  * @param {String} bucket_name name of S3 bucket
  * @param {String} key key name within bucket (including prefix)
- * @returns {String} string of dog or cat
+ * @returns {String} string of "Dog", "Cat" or "Neither"
  */
-const dogOrCat = (bucketName, key) => {
-  return "dog"
+const dogOrCat = (bucket_name, key) => {
+  const client = new RekognitionClient({ region: "us-east-1" });
+
+  const params = {
+    Image: {
+      S3Object: {
+        Bucket: bucket_name,
+        Name: key,
+      },
+    },
+    MinConfidence: 50,
+    Settings: {
+      GeneralLabels: {
+        LabelInclusionFilters: [
+          "Dog", "Cat"
+        ]
+      }
+    }
+  }
+  const command = new DetectLabelsCommand(params);
+  
+  client.send(command).then(
+    (data) => {
+      if (data.Labels.length > 0) {
+        console.log(`I am ${data.Labels[0].Confidence}% sure this is a ${data.Labels[0].Name}!`)
+        return data.Labels[0].Name
```

Comment:
Have you verified that this is actually being returned? I think this will only get "returned" if you "fetch" that from the client. F.e.: const result = await client....

Reply:
Not at my PC until later but I definitely see the log above. I cannot remember exactly if I saw the return statement actually return. But I did say I verified the whole lambda locally so I hope I did 🙃

Can check later

Reply:
Ahh `animals` was undefined but lambda was running through. Think I need to brush up on async